Made some changes on my side.

popcornell opened the pull request.

ddwkim committed:
Add Ksponspeech data preparation code

Co-authored-by: Dongwoo Arthur Kim <kimdwkimdw@gmail.com>

ddwkim committed:
Add KsponSpeech recipe for tokenizer

Co-authored-by: SeungMin Lucas Lee <lsm.somniator@gmail.com>

ddwkim committed:
Add KsponSpeech recipe for LM

Co-authored-by: Dongwoo Arthur Kim <kimdwkimdw@gmail.com>

ddwkim committed:
Add Ksponspeech recipe for ASR model

Co-authored-by: Dongwoo Arthur Kim <kimdwkimdw@gmail.com>

ddwkim committed:
KsponSpeech README.md updated, and pre-commit and KsponSpeech test added

Co-authored-by: Dongwoo Arthur Kim <kimdwkimdw@gmail.com>

aheba committed:
add statistical pooling (handling padding)

ycemsubakan committed:
removing the separate dynamic mixing file, using the one from WHAMandWHAMR recipe

ycemsubakan committed:
removing hard coded paths from the yaml file

ycemsubakan committed:
added comments to train.py

ycemsubakan committed:
fixed the comments in save_results function

aheba committed:
add dict of HF models/configs for better interface with HF

aheba committed:
fix linters

aheba committed:
update README.md

aheba committed:
close #1028

Gastron committed:
Pretrained subclasses nn.Module now

ycemsubakan committed:
updated README.md

ycemsubakan committed:
fixed variable name in the inference function

aheba committed:
update README

aheba committed:
fix typos

aheba committed:
add comment on IEMOCAP Regex

ycemsubakan committed:
added automatic preprocessing for whamr dynamic mixing

ycemsubakan committed:
added automatic preprocessing for whamr dynamic mixing

ycemsubakan committed:
removed hard coded links from train.py

Gastron committed:
SuperInit

aheba committed:
fix #1048

aheba committed:
add EncoderWav2vecClassifier

Gastron committed:
modules needs to be renamed

Gastron committed:
Modules to mods

mravanelli committed:
add inference repo in README

mravanelli committed:
Merge pull request #970 from BenoitWang/iemocap_wav2vec2

Benchmark fine-tuned wav2vec2 on IEMOCAP

mravanelli committed:
Update README.md

ycemsubakan committed:
fixing the path issue for wham_room file

ycemsubakan committed:
Merge branch 'snr_estimation' of github.com:speechbrain/speechbrain into snr_estimation

ycemsubakan committed:
added mixed precision training

mravanelli committed:
Merge pull request #1030 from speechbrain/snr_estimation

Neural SI-SNR Estimator for the REAL-M dataset

Gastron committed:
Fix for user site editable install

Gastron committed:
Merge branch 'develop' into pretrained-to-module

This also implements (sorry bout the ugly commit) some forward methods

Gastron committed:
Fix hparams->mods in SNREstimator

Gastron committed:
Add missing Hparams needed

TParcollet committed:
typo

mravanelli committed:
Merge pull request #1052 from Gastron/user-site-install

Fix for user site editable install

mravanelli committed:
Merge pull request #1042 from Gastron/pretrained-to-module

Pretrained subclasses nn.Module now

ycemsubakan committed:
added links for the REAL-M dataset and arxiv paper to the README

ycemsubakan committed:
added a citation section for REAL-M

mravanelli committed:
Merge pull request #1053 from speechbrain/sisnrest_readme

added links for the REAL-M dataset and arxiv paper to the README

mravanelli committed:
update requirements

Gastron committed:
Generic interface

ycemsubakan committed:
added the expansion to true SI-SNR range for the SI-SNR estimator

mravanelli committed:
Merge pull request #1059 from speechbrain/snrestimator_interfaceupdate

Added the expansion to true SI-SNR range for the SI-SNR estimator

aheba committed:
update setup with torch version 1.8

aheba committed:
Update setup.py

ferugit committed:
use the parameter layers to determine the number of RNN layers

Gastron committed:
Merge pull request #1069 from aheba/fix-pip-torch-version

Include torch version in setup.py

Gastron committed:
Add missing checkpoint hook register decoration

mravanelli committed:
Merge pull request #1073 from Gastron/fix/missing-ckpt-hook-cyclic-lr

Add missing checkpoint hook register decoration

AK391 committed:
updates

AK391 committed:
Merge branch 'speechbrain:develop' into develop

AK391 committed:
Delete demo.py

AK391 committed:
Update README.md

AK391 committed:
Update README.md

AK391 committed:
Update README.md

AK391 committed:
Update README.md

AK391 committed:
Update README.md

AK391 committed:
Update README.md

AK391 committed:
Update README.md

AK391 committed:
Update README.md

AK391 committed:
Update README.md

AK391 committed:
Update README.md

AK391 committed:
Update README.md

AK391 committed:
Update README.md

ferugit committed:
fix PEP8

ferugit committed:
fix PEP8

TParcollet committed:
Merge pull request #1071 from ferugit/fix-ASRfromScratch

Use the parameter layers to determine the number of RNN layers in templates/speech_recognition/LM/custom_model.py

EugKar committed:
Fixed freezing wav2vec at LibriSpeech CTC recipe

A fix of #1077

Gastron committed:
Remove torch.no_grad in features

TParcollet committed:
Merge pull request #1080 from EugKar/patch-1

Fixed freezing wav2vec at LibriSpeech CTC recipe

TParcollet committed:
Freeze wav2vec after request merged

JasonSWFu committed:
Metricgan-u SE update

mravanelli committed:
Merge pull request #1074 from AK391/develop

migrating gradio hub demo to huggingface spaces

JasonSWFu committed:
Metricgan-u dereverb update

JasonSWFu committed:
Metricgan-u SE update2

JasonSWFu committed:
Metricgan-u SE README update

hieuthi committed:
Fixed voicebank download error

torchaudio.save does not support 1D tensor anymore

JasonSWFu committed:
README update

JasonSWFu committed:
README update

ycemsubakan committed:
Added Multi-GPU usage instructions for separation recipes

mravanelli committed:
Merge pull request #1084 from speechbrain/sepformer_readmeupdate

Added Multi-GPU usage instructions for separation recipes

mravanelli committed:
Merge pull request #1081 from Gastron/grads-through-feature-lobes

Remove torch.no_grad in feature lobes

mravanelli committed:
Merge branch 'speechbrain:develop' into py1.10

mravanelli committed:
fixed some tests

ycemsubakan committed:
fixed device proble in separation resampling

mravanelli committed:
Merge pull request #1054 from mravanelli/py1.10

Support PyTorch 1.10

mravanelli committed:
Merge pull request #1083 from nii-yamagishilab/bug-voicebank-download-torchaudio

Fixed voicebank download error

mravanelli committed:
Merge pull request #1085 from speechbrain/inference_fix_separation

Fixed device problem in separation resampling

mravanelli committed:
fix linters

Gastron committed:
Clarify Readme

Clarified how the WebDataset integration really works and corrected what the NFS acronym stands for.

mravanelli committed:
Update test_normalization.py

mravanelli committed:
Update test_features.py

mravanelli committed:
fix torchaudio backend issue

Commit:
Fix minor typo

Gastron committed:
Add default checkpoint hooks for Torch Schedulers

mravanelli committed:
Merge pull request #1087 from mravanelli/backend_fix

Torchaudio backed

mravanelli committed:
Merge pull request #1091 from Gastron/torch-schedulers-default-ckpt-hooks

Add default checkpoint hooks for Torch Schedulers

underdogliu committed:
change torchaudio minimal version for compatibility

mravanelli committed:
Merge pull request #1094 from underdogliu/develop

change torchaudio minimal version for compatibility

FrancoisGrondin committed:
Fix MVDR

mravanelli committed:
Merge pull request #1095 from FrancoisGrondin/mvdr

Fix MVDR

Gastron committed:
Pass file as string

mravanelli committed:
Merge pull request #1102 from Gastron/fix/pass-str-to-torchaudio

Pass file as string

mravanelli committed:
Update README.md

flexthink committed:
Add convenience wrappers for hyperparameter fitting via Orion

flexthink committed:
Add the ability to avoid loading the pretrained model

vitas-ai committed:
original recipe (before was the saved one of the results folder)

vitas-ai committed:
placeholders and such...

vitas-ai committed:
edit entry & the used CV path for 7.0-de

TParcollet committed:
remove CUDNN flags

TParcollet committed:
Remove True value for argument

mravanelli committed:
Merge pull request #1111 from TParcollet/fix_ddp_conv

remove CUDNN flags to fix DDP with dilated conv

mravanelli committed:
fix device mismatch in multimic processing for supporting gpu use

mravanelli committed:
fix other device mismatch in multi_mic

mravanelli committed:
fix test

mravanelli committed:
fix test

mravanelli committed:
add len in enhance_file interface

mravanelli committed:
fix test

mravanelli committed:
fix last test

weiwei-ww committed:
Fix the padding issue

flexthink committed:
Update exception handling for hyperparameter fitting

flexthink committed:
Add a speaker-id template clone for a hyperparameter fitting demo

mravanelli committed:
fix devie mismatch

flexthink committed:
Added a speaker-id example, updated hpfit with the ability to save the model for each run.

flexthink committed:
Address yamllint issues

flexthink committed:
Delete the original ASR example

popcornell committed:
Merge branch 'develop' of https://github.com/speechbrain/speechbrain into dynamic

popcornell committed:
minimizing args, keeping Andreas solution.
Adding max_batch_ex to limit batches with too many examples (suggested by Titouan).

flexthink committed:
Update formatting (for linters)

popcornell committed:
need to add max_batch_ex into _reduce_padding: ask Andreas

popcornell committed:
minor changes to core.py.

popcornell committed:
use optionally dynamic batching

popcornell committed:
merged dynamic_batching training script in to train.py

popcornell committed:
merged dynamic_batching training script in to train.py

popcornell commented:
@vitas-ai  can you pull from speechbrain/develop so this PR has less clutter ?

amrit110 committed:
Add ctc-segmentation to mocked deps

Add ctc-segmentation package
(https://github.com/lumaku/ctc-segmentation) to be mocked while
generating API docs using sphinx autodoc.

amrit110 committed:
Add ctc-segmentation to mocked deps

Add ctc-segmentation package
(https://github.com/lumaku/ctc-segmentation) to be mocked while
generating API docs using sphinx autodoc.

amrit110 committed:
Merge branch 'mock_ctc_segmentation' of github.com:amrit110/speechbrain into mock_ctc_segmentation

BenoitWang committed:
add wav2vec2 encoder

amrit110 committed:
Update docs requirements

Update docs requirements file to add ctc-segmentation package
Update docs requirements to have numba version minimum
Remove autodoc mock imports for numba

popcornell committed:
updated tests

popcornell committed:
Merge branch 'speechbrain:develop' into padding

popcornell committed:
Merge branch 'develop' of https://github.com/speechbrain/speechbrain into padding

popcornell committed:
Merge remote-tracking branch 'popcornell/padding' into padding

weiwei-ww committed:
Merge branch 'popcornell-padding' into padding

Gastron committed:
Merge pull request #1127 from amrit110/update_docs_requirements

Update docs requirements

mravanelli committed:
fixed comment

mravanelli committed:
Merge pull request #1115 from weiwei-ww/padding

Fix the padding issue

mravanelli committed:
Update convolution.py

mravanelli committed:
fix test

ddwkim committed:
Update citation and ASR decoding hyperparameters

BenoitWang committed:
update readme

kimdwkimdw committed:
Simplify .github workflow

kimdwkimdw committed:
Add pytest to only for pytest

kimdwkimdw committed:
Apply pre-commit all

Gastron committed:
Automatically read from sys, fix some docstrings

Gastron committed:
Document hparams overriding for run_opts

kimdwkimdw committed:
divide pre-commit.yml

Gastron committed:
Merge develop and add docstring

mravanelli committed:
update gdrive link

mravanelli committed:
update HF link

mravanelli committed:
Merge pull request #973 from kimdwkimdw/recipe/ksponspeech

Add KsponSpeech recipe

mravanelli committed:
Merge pull request #1135 from Gastron/better-argparsing

Document hparams overriding of run_opts and automatically load argv[1:] if not specified.

mravanelli committed:
add examples in doctring, merged MetricGAN_U_dnsmos.py and MetricGAN_U_srmr.py (were identical), update readme

mravanelli committed:
fix linters

mravanelli committed:
fix BLSTM.yaml + make model directory in hparams for both recipes

mravanelli committed:
update extra-requirements

JasonSWFu committed:
Fix some bugs(stability) of metricgan+

JasonSWFu committed:
Merge branch 'Metricgan-u' of https://github.com/JasonSWFu/speechbrain into Metricgan-u

pull

JasonSWFu committed:
Fix some changes in ./enhance/MetricGAN-U

vitas-ai commented:
I did that, hence there is so much clutter ;-)
The dynamic_libri branch was outdated, so I fetched from dev and we set the PR to dev in the speechbrain repo (instead of to the dynamic_libri branch).

BenoitWang committed:
Update train_with_wav2vec2.py

BenoitWang committed:
Update train_with_wav2vec2.py

BenoitWang committed:
Update train.py

BenoitWang committed:
Update train_with_wav2vec2.py

Gastron committed:
Auto mix prec + noprogressbar override via hyperparams

BenoitWang committed:
Update train_with_wav2vec2.yaml

mravanelli committed:
last fixes, including setting n_jobs=1 by default (sometimes with >1 do not work on some nodes)

mravanelli committed:
add gdrive link

ycemsubakan committed:
Merge pull request #1140 from Gastron/auto-mix-prec-no-default

Auto mix prec + noprogressbar override via hyperparams

flexthink committed:
Rename hyperparameter fitting (hpfit) to hyperparameter optimization (hpopt)

mravanelli committed:
Update README.md

mravanelli committed:
Update README.md

mravanelli committed:
Update README.md

mravanelli committed:
Merge pull request #1120 from JasonSWFu/Metricgan-u

Metricgan-U

mravanelli committed:
fix imprecise .int() conversion when computing abs len

mravanelli committed:
Merge pull request #1144 from speechbrain/fixround

fix imprecise .int() conversion

mravanelli committed:
Update installation.md

mravanelli committed:
Merge pull request #1122 from flexthink/orion

Hyperparameter Fitting Support

Gastron committed:
Custom python module can be used for any pretrained

Gastron committed:
Documentation and safeguards

mravanelli committed:
allowing tests to run on cuda as well

mravanelli committed:
update installation.md with cuda tests

mravanelli committed:
Merge pull request #1078 from vitas-ai/develop

DE ASR model (seq2seq) from CommonVoice7.0

popcornell committed:
do not use bucket bounds by default

popcornell committed:
speeding up reduce_padding

mravanelli committed:
Merge pull request #1146 from mravanelli/pytest-cuda

Pytest on cuda

kimdwkimdw committed:
Rollback lint-requirements.txt

BenoitWang committed:
correct when decodage not dict

BenoitWang committed:
Update train_with_wav2vec2.py

BenoitWang committed:
correct when decodage not dict

BenoitWang committed:
Update train_with_wav2vec2.py

kimdwkimdw committed:
Apply pre-commit on latest develop changes

aheba committed:
remove EncoderWav2vecClassifier

Gastron committed:
Merge pull request #1134 from kimdwkimdw/feature/pre-commit_workflow

Simplify .github workflow using pre-commit

aheba committed:
Merge pull request #1058 from Gastron/generic-pretrained-interface

Generic Pretrained interface

popcornell committed:
removed reduce_padding for now

popcornell committed:
changed seq2seq recipe accordingly

popcornell committed:
changed TIMIT recipe accordingly

Moumeneb1 committed:
add_dummy_prefix arg

Add the possibility to control the dummy prefix in sentence piece tokenizer. like in https://github.com/google/sentencepiece/issues/305

BenoitWang committed:
Update train_with_wav2vec2.py

BenoitWang committed:
delete env_corrupt

BenoitWang committed:
Update train_with_wav2vec2.py

BenoitWang committed:
Update train_with_wav2vec2.yaml

BenoitWang committed:
Update train_with_wav2vec2.py

aheba committed:
fix linters

aheba committed:
Merge pull request #1152 from Moumeneb1/add_dummy_prefix

add_dummy_prefix arg

ferugit committed:
make ctc-segmentation work with EncoderASR

aheba committed:
keep the default value: True

ferugit committed:
fix pre-commit test

ferugit committed:
fix pre-commit test

BenoitWang committed:
add model link

ferugit committed:
replace decoder by encoder and correct usage of log_softmax

Gastron committed:
Remove ddp barrier from intra-epoch ckpt saving

TParcollet committed:
Merge pull request #1154 from Moumeneb1/add_dummy_prefix

SentencePiece add_dummy_prefix - keep the default value

TParcollet committed:
Update README.md

TParcollet committed:
Update README.md

TParcollet committed:
Update README.md

BenoitWang committed:
Add results link

BenoitWang committed:
Add results link

BenoitWang committed:
Add results link

BenoitWang committed:
Add results link

aheba committed:
Merge pull request #1167 from Gastron/fix/ckpt-ddp-barrier

Remove ddp barrier from intra-epoch ckpt saving

aheba committed:
add transformers dep

aheba committed:
remove unused env_corrupt module

aheba committed:
switch tokenizer from dropbox to HF download

aheba committed:
merge publi ref

mravanelli committed:
Merge pull request #1156 from ferugit/improve-ctc-segmentation

make ctc-segmentation work with EncoderASR

aheba committed:
Merge pull request #1139 from BenoitWang/slurp_wav2vec2

Benchmark fine-tuned wav2vec2/HuBERT on SLURP

popcornell committed:
Merge branch 'develop' of https://github.com/speechbrain/speechbrain into dynamic

 Conflicts:
	recipes/CommonVoice/ASR/seq2seq/README.md
	recipes/CommonVoice/ASR/seq2seq/hparams/train_de.yaml

popcornell committed:
docs changes

popcornell committed:
change to doctring and doctest

popcornell committed:
removed reduce_padding

popcornell committed:
verbose to False by default

mravanelli committed:
fix missing keys

popcornell committed:
removed max_batch_ex from recipe
